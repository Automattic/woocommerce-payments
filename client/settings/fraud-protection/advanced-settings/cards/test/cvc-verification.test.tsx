/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';
/**
 * Internal dependencies
 */
import FraudPreventionSettingsContext from '../../context';
import CVCVerificationRuleCard from '../cvc-verification';

declare const global: {
	wcpaySettings: {
		accountStatus: {
			fraudProtection: {
				declineOnCVCFailure: boolean;
			};
		};
		isFRTReviewFeatureActive?: boolean;
	};
};

describe( 'CVC verification card', () => {
	test( 'renders correctly when CVC check is enabled', () => {
		const settings = {
			cvc_verification: {
				enabled: false,
				block: false,
			},
		};
		global.wcpaySettings = {
			accountStatus: {
				fraudProtection: {
					declineOnCVCFailure: true,
				},
			},
			isFRTReviewFeatureActive: false,
		};
		const setSettings = jest.fn();
		const contextValue = {
			protectionSettingsUI: settings,
			setProtectionSettingsUI: setSettings,
			protectionSettingsChanged: false,
			setProtectionSettingsChanged: jest.fn(),
			setHasChanges: jest.fn(),
		};
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<CVCVerificationRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
		expect( container ).toHaveTextContent(
			/For security, this filter is enabled and cannot be modified/i
		);
	} );
	test( 'renders correctly when CVC check is disabled', () => {
		const settings = {
			cvc_verification: {
				enabled: false,
				block: false,
			},
		};
		global.wcpaySettings = {
			accountStatus: {
				fraudProtection: {
					declineOnCVCFailure: false,
				},
			},
		};
		const setSettings = jest.fn();
		const contextValue = {
			protectionSettingsUI: settings,
			setProtectionSettingsUI: setSettings,
			protectionSettingsChanged: false,
			setProtectionSettingsChanged: jest.fn(),
			setHasChanges: jest.fn(),
		};
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<CVCVerificationRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
		expect( container ).toHaveTextContent(
			/This filter is disabled, and can not be modified/i
		);
	} );
} );
