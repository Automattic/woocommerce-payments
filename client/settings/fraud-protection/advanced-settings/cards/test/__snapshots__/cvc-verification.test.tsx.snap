// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVC verification card renders correctly when CVC check is disabled 1`] = `
<div>
  <div
    class="components-surface components-card fraud-protection-rule-card css-nsno0f-View-Surface-getBorders-primary-Card-rounded em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Card"
    id="cvc-verification-card"
  >
    <div
      class="css-mgwsf4-View-Content em57xhy0"
    >
      <div
        class="components-card__body components-card-body wcpay-card-body fraud-protection-rule-card-header-container css-1nwhnu3-View-Body-borderRadius-medium em57xhy0"
        data-wp-c16t="true"
        data-wp-component="CardBody"
      >
        <div>
          <p
            class="fraud-protection-rule-card-header"
          >
            CVC Verification
          </p>
          <p
            class="fraud-protection-rule-card-description"
          >
            This filter checks the security code submitted by the customer against the data on file with the card issuer.
          </p>
        </div>
      </div>
      <hr />
      <div
        class="components-card__body components-card-body wcpay-card-body css-1nwhnu3-View-Body-borderRadius-medium em57xhy0"
        data-wp-c16t="true"
        data-wp-component="CardBody"
      >
        <div
          class="fraud-protection-rule-description"
        >
          <strong>
            How does this filter protect me?
          </strong>
          <p>
            Because the card security code appears only on the card and not on receipts or statements, the card security code provides some assurance that the physical card is in the possession of the buyer.
          </p>
        </div>
        <div
          class="wcpay-inline-notice wcpay-inline-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
        >
          <div
            class="components-notice__content"
          >
            <div
              class="components-flex css-bmzg3m-View-Flex-sx-Base-sx-Items-ItemsRow em57xhy0"
              data-wp-c16t="true"
              data-wp-component="Flex"
            >
              <div
                class="components-flex-item wcpay-inline-notice__icon wcpay-inline-warning-notice__icon css-mw3lhz-View-Item-sx-Base em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <svg
                  class="gridicon gridicons-notice-outline"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                    />
                  </g>
                </svg>
              </div>
              <div
                class="components-flex-item wcpay-inline-notice__content wcpay-inline-warning-notice__content css-mw3lhz-View-Item-sx-Base em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                This filter is disabled, and can not be modified.
              </div>
            </div>
            <div
              class="components-notice__actions"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
    <div
      aria-hidden="true"
      class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
  </div>
</div>
`;

exports[`CVC verification card renders correctly when CVC check is enabled 1`] = `
<div>
  <div
    class="components-surface components-card fraud-protection-rule-card css-nsno0f-View-Surface-getBorders-primary-Card-rounded em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Card"
    id="cvc-verification-card"
  >
    <div
      class="css-mgwsf4-View-Content em57xhy0"
    >
      <div
        class="components-card__body components-card-body wcpay-card-body fraud-protection-rule-card-header-container css-1nwhnu3-View-Body-borderRadius-medium em57xhy0"
        data-wp-c16t="true"
        data-wp-component="CardBody"
      >
        <div>
          <p
            class="fraud-protection-rule-card-header"
          >
            CVC Verification
          </p>
          <p
            class="fraud-protection-rule-card-description"
          >
            This filter checks the security code submitted by the customer against the data on file with the card issuer.
          </p>
        </div>
      </div>
      <hr />
      <div
        class="components-card__body components-card-body wcpay-card-body css-1nwhnu3-View-Body-borderRadius-medium em57xhy0"
        data-wp-c16t="true"
        data-wp-component="CardBody"
      >
        <div
          class="fraud-protection-rule-description"
        >
          <strong>
            How does this filter protect me?
          </strong>
          <p>
            Because the card security code appears only on the card and not on receipts or statements, the card security code provides some assurance that the physical card is in the possession of the buyer.
          </p>
        </div>
        <div
          class="wcpay-inline-notice wcpay-inline-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
        >
          <div
            class="components-notice__content"
          >
            <div
              class="components-flex css-bmzg3m-View-Flex-sx-Base-sx-Items-ItemsRow em57xhy0"
              data-wp-c16t="true"
              data-wp-component="Flex"
            >
              <div
                class="components-flex-item wcpay-inline-notice__icon wcpay-inline-warning-notice__icon css-mw3lhz-View-Item-sx-Base em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <svg
                  class="gridicon gridicons-notice-outline"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                    />
                  </g>
                </svg>
              </div>
              <div
                class="components-flex-item wcpay-inline-notice__content wcpay-inline-warning-notice__content css-mw3lhz-View-Item-sx-Base em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                For security, this filter is enabled and cannot be modified. Payments failing CVC verification will be blocked.
                <a
                  data-link-type="external"
                  href="https://woo.com/document/woopayments/fraud-and-disputes/fraud-protection/#advanced-configuration"
                  target="_blank"
                >
                  Learn more
                </a>
              </div>
            </div>
            <div
              class="components-notice__actions"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
    <div
      aria-hidden="true"
      class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
  </div>
</div>
`;
